### PR TITLE
Add AddAppCampaign example

### PR DESCRIPTION
**Pull request by devchas**

### CODE REVIEW

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddAppCampaign.java [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+            // manual_cpc, commission, maximize_conversions, etc.
+            // See https://developers.google.com/google-ads/api/reference/rpc
+            // under current version / resources / Campaign.
+            .setTargetCpa(TargetCpa.newBuilder().setTargetCpaMicros(Int64Value.of(1000000)).build())
+            // Sets the App campaign settings.
+            .setAppCampaignSetting(
+                AppCampaignSetting.newBuilder()
+                    .setAppId(StringValue.of("com.google.android.apps.adwords"))
+                    .setAppStore(AppCampaignAppStore.GOOGLE_APP_STORE)
+                    .setBiddingStrategyGoalType(
```
Comment (PierrickVoulet) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddAppCampaign.java]: A comment is missing "Optional: Optimize this campaign for getting new users for your app."

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddAppCampaign.java [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+      // Prints and returns the campaign resource name.
+      String campaignResourceName = response.getResults(0).getResourceName();
+      System.out.printf("Created App campaign with resource name '%s'.%n", campaignResourceName);
+      return campaignResourceName;
+    }
+  }
+
+  /**
+   * Sets campaign targeting criteria for a given campaign.
+   *
```
Comment (PierrickVoulet) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddAppCampaign.java]: Missing a comment "Both location and language targeting are illustrated."

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddAppCampaign.java [hunk 3]
```diff
@@ -0,0 +1,427 @@
...
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the customer ID.
+   * @param adGroupResourceName the resource name of the ad group to add the App ad to.
+   */
+  private void createAppAd(
+      GoogleAdsClient googleAdsClient, long customerId, String adGroupResourceName) {
+    // Creates an ad group ad.
+    AdGroupAd adGroupAd =
+        AdGroupAd.newBuilder()
+            .setAdGroup(StringValue.of(adGroupResourceName))
```
Comment (PierrickVoulet) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddAppCampaign.java]: "status" field to "ENABLED" seems to be missing?

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddAppCampaign.java [hunk 4]
```diff
@@ -0,0 +1,432 @@
...
+import com.google.protobuf.Int64Value;
+import com.google.protobuf.StringValue;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.joda.time.DateTime;
+
+/** Adds a new App Campaign */
```
Comment (jradcliff) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddAppCampaign.java]: nit: add period at end

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddAppCampaign.java [hunk 5]
```diff
@@ -0,0 +1,432 @@
...
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the customer ID.
+   * @return the resource name of the newly created campaign budget.
+   */
+  private String createBudget(GoogleAdsClient googleAdsClient, long customerId) {
+    // Creates a campaign budget.
+    CampaignBudget campaignBudget =
+        CampaignBudget.newBuilder()
+            .setName(StringValue.of("Interplanetary Cruise #" + System.currentTimeMillis()))
+            .setAmountMicros(Int64Value.of(50000000))
```
Comment (jradcliff) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddAppCampaign.java]: Please write as `50_000_000` for readability

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddAppCampaign.java [hunk 6]
```diff
@@ -0,0 +1,432 @@
...
+        CampaignBudgetOperation.newBuilder().setCreate(campaignBudget).build();
+
+    // Creates the campaign budget service client.
+    try (CampaignBudgetServiceClient campaignBudgetServiceClient =
+        googleAdsClient.getLatestVersion().createCampaignBudgetServiceClient()) {
+      // Adds the campaign budget.
+      MutateCampaignBudgetsResponse response =
+          campaignBudgetServiceClient.mutateCampaignBudgets(
+              Long.toString(customerId), ImmutableList.of(operation));
+      // Prints and returns the response.
```
Comment (jradcliff) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddAppCampaign.java]: nit: s/response/resource name

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddAppCampaign.java [hunk 7]
```diff
@@ -0,0 +1,432 @@
...
+  private void setCampaignTargetingCriteria(
+      GoogleAdsClient googleAdsClient, long customerId, String campaignResourceName) {
+    List<CampaignCriterionOperation> operations = new ArrayList<>();
+
+    // Creates the location campaign criteria.
+    // Besides using location_id, you can also search by location names from
+    // GeoTargetConstantService.suggestGeoTargetConstants() and directly
+    // apply GeoTargetConstant.resource_name here. An example can be found
+    // in GetGeoTargetConstantByNames.java.
+    ArrayList<Integer> locationIds =
```
Comment (jradcliff) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddAppCampaign.java]: You can simplify this to:

```
List<Integer> locationIds = Arrays.asList( ... );
```

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddAppCampaign.java [hunk 8]
```diff
@@ -0,0 +1,432 @@
...
+
+      // Creates a campaign criterion operation.
+      CampaignCriterionOperation operation =
+          CampaignCriterionOperation.newBuilder().setCreate(campaignCriterion).build();
+
+      operations.add(operation);
+    }
+
+    // Creates the language campaign criteria.
+    ArrayList<Integer> languageIds =
```
Comment (jradcliff) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddAppCampaign.java]: See above

**Pull request merged**